src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,53 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter)
+                    && (previouslyMatchedEpisodes == other.previouslyMatchedEpisodes);
         }
     };
 }
 
 using namespace RSS;
 
+static const QString episodeRegex = "(?:^|[^A-Za-z0-9])((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))(?:[^A-Za-z0-9]|$)";
+
+QString computeEpisodeName(const QString &article)
+{
+    QRegExp f(episodeRegex, Qt::CaseInsensitive);
+    // See if we can extract an season/episode number or date from the title
+    if (f.indexIn(article) == -1) {
+        return QString();
+    }
+
+    // Extract out the episode identifier
+    QString matchedEpisode = f.cap(1);
+
+    // Is it in the S01E01 format?
+    if (matchedEpisode.startsWith('S', Qt::CaseInsensitive)) {
```

Comment:
Well that's unfortunate. That being the case, how about one of these options?
1. Use captured groups for each component and identify the match by the `lastCapturedIndex()`. Since you are always capturing in pairs, episode should always be `lastCapturedIndex()` and season `lastCapturedIndex() - 1` 
2. Use an array of patterns using named captures and take the first that matches. This would likely be the cleanest in terms of code readability and maintenance , but has a high overhead.